src/include/miopen/logger.hpp

```diff
@@ -383,6 +408,31 @@ class LogScopeTime
 #define MIOPEN_LOG_SCOPE_TIME
 #endif
 
+class LogScopeRoctx
+{
+public:
+    LogScopeRoctx() {};
+    LogScopeRoctx(const std::string &name)
```

Comment:
This is the autolocker/scoped-marker pattern.  We SHOULD have passed the scope name into the constructor .  See the LogScopeTime class right above.  The "empty constructor/pass the name in later" is the hack that lets up keep the marker class in the stack-frame that called the logging macro.

Just because we are using the hack 100% of the time doesn't mean we shouldn't have the proper constructor. :)